src/core/datastore/sync/sync-manager.js

```diff
@@ -181,20 +192,45 @@ export class SyncManager {
     return this._networkRepo.update(collection, entity)
       .then((updateResult) => {
         result.entity = updateResult;
-        return result;
+        return this._getOfflineRepo();
       })
+      .then(repo => repo.update(collection, result.entity))
+      .then(() => result)
       .catch((err) => {
         result.entity = entity;
         result.error = err;
         return result;
       });
   }
 
-  // TODO: refactor results of individual push ops can be done here?
-  _pushItem({ collection, entityId, state }) {
-    let entity;
+  _handlePushOp(syncItem, offlineEntity) {
+    const { collection, state, entityId } = syncItem;
     const syncOp = state.operation;
 
+    if (!offlineEntity && syncOp !== SyncOperation.Delete) { // todo: duplication
+      const res = this._getPushOpResult(entityId, syncOp);
+      res.error = new KinveyError(`Entity with id ${entityId} not found`);
```

Comment:
What I was going for was backwards compatibility.

Whether we should delete it... I guess we could. I'm usually not comfortable making decisions for the user, but there isn't much more one can do, than deleting it. And it is somewhat cumbersome to find and handle these errors as a user. I think we can delete the sync item.

Reply:
Agreed re: backwards compatibility. I realize you were only aiming for feature parity.
These reviews are a good opportunity for me to review our current behavior in more depth - hence the question :)

This particular question stems from recent bug reports where a developer got into a state of "pending push items that fail when pushed". The case being handled in your code here is not exactly that, but the outcome will be the same - the developer is stuck at that point with a corrupt cache.
I agree that we should avoid making decisions on behalf of the developer. In this case, though, the developer is completely out of options. As much as we can, let's keep the state of the cache consistent. 

Feel free to add the improvement here or separately.

Reply:
I think it would be best to add it as part of [MLIBZ-2274](https://kinvey.atlassian.net/browse/MLIBZ-2274), to avoid messing up the chain of PRs now currently open, any further.

I'm adding a comment to the above mentioned ticket about this with a bit of a "spec" :)